apps/designer/app/designer/features/style-panel/sections/spacing/types.ts

```diff
@@ -0,0 +1,17 @@
+import type { StyleValue, StyleProperty } from "@webstudio-is/css-data";
+import { categories } from "@webstudio-is/react-sdk";
+
+export type StyleChangeEvent = {
```

Comment:
hmm, why do we need a separate type for spacing, isn't this something we already have for other things?

Reply:
This type allowed me to unify nicely all change handlers in spacing. In other places we use two callbacks to distinguish between `isEphemeral` and we don't pass `property`.

Maybe we should use this type everywhere not sure

Reply:
Yeah, if anything consistency is important

Reply:
We got StyleUpdate and StyleUpdateOptions. 
Options is used locally in the style panel only, while StyleUpdate represents datastructure that goes beyond style panel, that's why it was separated

Reply:
if every panel changes these things it will be harder to maintain, so lets have one thing, whatever that is

Reply:
Using StyleUpdate and StyleUpdateOptions now